modules/json-schema/test/src/AllOfSpec.scala

```diff
@@ -144,4 +144,116 @@ final class AllOfSpec extends munit.FunSuite {
+         |            "type" : "integer"
+         |         }
+         |      }
+         |    }
+         |  }
+         |}
+         |""".stripMargin
+
+    val expectedString = """|namespace foo
+                            |structure Example with [Two] {}
+                            |
+                            |structure Test {
+                            |  example: Example
+                            |}
+                            |
+                            |@mixin
+                            |structure Two {
+                            |  vehicle: String,
+                            |  price: Integer
+                            |}
+                            |
+                            |""".stripMargin
+
+    TestUtils.runConversionTest(jsonSchString, expectedString)
+  }
+
+  test("unions - Parent type explicitly referenced and extended".only) {
+    val jsonSchString =
+      """|{
+         |  "$id": "test.json",
+         |  "$schema": "http://json-schema.org/draft-07/schema#",
+         |  "title": "Test",
+         |  "type" : "object",
+         |  "properties" : {
+         |    "root": { "$ref": "#/$defs/root" }
+         |  },
+         |  "$defs": {
+         |    "root": { 
+         |      "type": "object",
+         |      "properties": { 
+         |        "s": { 
+         |          "type": "string"
+         |        }
+         |      }
+         |    },
+         |    "subB": {
+         |      "type": "object",
+         |      "allOf": [
+         |        { "$ref": "#/$defs/root" },
+         |        { "type": "object", "properties": { "x": { "type": "integer" } } }
+         |      ]
+         |    },
+         |    "subA": {
+         |      "type": "object",
+         |      "allOf": [
+         |        { "$ref": "#/$defs/root" },
+         |        { "type": "object", "properties": { "x": { "type": "integer" } } }
+         |      ]
+         |    }
+         |  }
+         |}
+         |""".stripMargin
+
+    val expectedString = """|namespace foo
+                            |
+                            |structure Root with [RootMixin] {} 
+                            |
+                            |@mixin
+                            |structure RootMixin {
+                            |  s: String
+                            |}
+                            |
+                            |structure SubA with [RootMixin] {
+                            |  x: Integer
+                            |}
+                            |
+                            |structure SubB with [RootMixin] {
+                            |  x: Integer 
+                            |}
+                            |
+                            |structure Test {
+                            |  root: Root
+                            |}
+                            |
+                            |""".stripMargin
+
+    TestUtils.runConversionTest(jsonSchString, expectedString)
+  }
```

Comment:
This test is inline with what we would expect from current behavior, but it opens up some new issues. 

1. This test currently fails. The parent type (root) is not having any field members generated
2. Should the `Test` structure actually be referencing `RootMixin`?  In the JsonSchema I am looking at, the parent type is referenced with the expectation that any sub-type can take its place.  The current implementation does not allow that, because `SubA` and `SubB` are sub-types of `RootMixin`, and not `Root`.

Reply:
Looking at the test case above, I don't see anything in the `expectedString` smithy that seems wrong or out of place to me, perhaps you can provide an example of what you would think it should be as opposed to the current and that will help me understand what you mean